tools/config.py

```diff
@@ -797,6 +798,13 @@ def validate_config(self):
         return True
 
 
+    @property
+    def name(self):
+        if "artifact_name" in self.app_config_data:
```

Comment:
Strangely enough, you can put an mbed_app.json in the root of a library build folder and it will be picked up.